### PR TITLE
ISS-189649:  Update the mobile SDK documentation

### PR DESCRIPTION
**Pull request by dhananjayj-dev**
## Summary

This PR updates the public documentation to reflect the newly introduced custom masking architecture in the DevRev SDK.
It also fixes a number of inconsistencies across platforms.

## Connected Issues

- [ISS-189649](https://app.devrev.ai/devrev/works/ISS-189649)

## Craftsmanship, Integrity, and Devil’s Advocacy

- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
- [ ] Observing: Detailed error codes so machines can understand
- [ ] Observing: Adding superior metrics for future debugging
- [ ] Observing: Tracing the hairiest pathways for field serviceability
- [ ] Training: KnowledgeOps update? So AI always works.

### CODE REVIEW

File: fern/docs/pages/sdks/mobile/flutter/quickstart.mdx
[the remark this answers is not in the document]
Comment (dhananjayj-dev) [reply fern/docs/pages/sdks/mobile/flutter/quickstart.mdx:1]: At other places there is an empty line.

**Comment by dhananjayj-dev** [comment]
@bc-devrev @bojan - Just FYI, temporarily I have used `code` icon as the original flutter icon issue is raised and it will probably be resolved next week.
Screenshot 2025-08-01 at 1 27 18 PM

File: fern/docs/pages/sdks/mobile/android/features.mdx [hunk 1]
```diff
@@ -630,6 +630,74 @@ For example:
 <input type="text" placeholder="Enter Username" name="username" required class="devrev-unmask">
 ```
 
+### Custom masking provider
+
+For advanced use cases, you can provide a custom masking provider to explicitly specify which regions of the UI should be masked during snapshots.
+
+You can implement your own masking logic by creating a class that implements the `MaskLocationProvider` interface and setting it via `DevRev.setMaskLocationProvider(MyMaskingProvider())`. This allows you to define exact areas to mask or skip snapshots entirely for privacy and security.
```
Comment (bojan) [inline fern/docs/pages/sdks/mobile/android/features.mdx:637]: We should use signature methods not implementations.

File: fern/docs/pages/sdks/mobile/flutter/quickstart.mdx
[the remark this answers is not in the document]
Comment (bojan) [reply fern/docs/pages/sdks/mobile/flutter/quickstart.mdx:1]: I checked everything, you are correct. Resolving.

**Review by bojan — CHANGES_REQUESTED** [review]

**Review by bc-devrev — APPROVED** [review]
Approved but please address the use of future tense before merging. https://www.notion.so/devrev/Why-should-I-avoid-future-tense-540a9f3531594769a1f686fdf2f4f9cf?source=copy_link

**Comment by dhananjayj-dev** [comment]
@bojan - I’ve added documentation for token exchange and identification, updates related to push notifications in the Cordova sample app and addressed the future tense occurrences. Please re-review it.

**Review by bojan — APPROVED** [review]
Can we just make sure we bring the changes to the readmes?

**Review by bojan — APPROVED** [review]

File: fern/docs/pages/sdks/mobile/android/features.mdx [hunk 2]
```diff
@@ -67,16 +67,67 @@ For example:
...
-
 ### Verified identification
 
-The verified identification method is used to identify the user with a unique identifier and verify the user's identity with the DevRev backend.
+The verified identification method is used to identify users with an identifier unique to your system within the DevRev platform. The verification is done through a token exchange process between you and the DevRev backend.
+
+The steps to identify a verified user are as follows:
+1. Generate an AAT for your system (preferably through your backend).
+2. Exchange your AAT for a session token for each user of your system.
+3. Pass the user identifier and the exchanged session token to the `DevRev.identifyVerifiedUser(_:sessionToken:)` method.
```
Comment (rohan-devrev) [inline fern/docs/pages/sdks/mobile/android/features.mdx:83]: This looks like an iOS method signature. `DevRev.identifyVerifiedUser(sessionToken: String)` can be used instead.

File: fern/docs/pages/sdks/mobile/android/features.mdx [hunk 3]
```diff
@@ -135,7 +186,80 @@ You can perform a logout of the current user by calling the following method:
   </Tab>
 </Tabs>
 
-The user will be logged out by clearing their credentials, as well as unregistering the device from receiving push notifications, and stopping the session recording.
+The user is logged out by clearing their credentials, as well as unregistering the device from receiving push notifications, and stopping the session recording.
+
+### Identity model
+
+The `Identity` class is used to provide user, organization, and account information when identifying users or updating their details. This class is used primarily with the `identifyUnverifiedUser(_:)` and `updateUser(_:)` methods.
```
Comment (rohan-devrev) [inline fern/docs/pages/sdks/mobile/android/features.mdx:193]: The signatures look off here as well.

File: fern/docs/pages/sdks/mobile/android/quickstart.mdx [hunk 4]
```diff
@@ -2,10 +2,11 @@
...
 
-- Android Studio 2022.1.1 or later
-- Android Gradle plugin version 7.4 or later
-- Gradle version 7.6 or later
-- Minimum Android SDK 24
+- Android Studio 2025.1.1 or later.
+- Android Gradle Plugin 8.2 or later.
+- Gradle 8.9 or later.
+- The minimum Android API level should be 24.
+- Recommended: An SSH key configured locally and registered with [GitHub](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).
```
Comment (rohan-devrev) [inline fern/docs/pages/sdks/mobile/android/quickstart.mdx:9]: None of our dependencies are linked with GitHub or have such prerequisites, so this may be redundant for Android.

**Review by rohan-devrev — CHANGES_REQUESTED** [review]

**Review by rohan-devrev — APPROVED** [review]

**Review by zeeshan-devrev — APPROVED** [review]

File: fern/docs/pages/sdks/mobile/react-native/features.mdx [hunk 5]
```diff
@@ -183,21 +314,34 @@ The mechanism uses balanced start and stop methods, both of which accept a timer
...
 To stop a timer, use the following method:
 
 ```typescript
-DevRev.stopTimer(name: string, properties: Map<string, string>)
+DevRev.endTimer(name: string, properties: { [key: string]: string })
 ```
 
 ### Screen tracking
 
 The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers are automatically tracked, you can manually track screens using the following method:
```
Comment (zeeshan-devrev) [inline fern/docs/pages/sdks/mobile/react-native/features.mdx:328]:
```suggestion
The DevRev SDK offers automatic screen tracking to help you understand how users navigate through your app. Although view controllers or activities are automatically tracked, you can manually track screens using the following method:
```

Comment (zeeshan-devrev) [inline fern/docs/pages/sdks/mobile/react-native/features.mdx:328]: I believe it would be more accurate to refer to this as `screen name tracking` rather than `screen tracking`, as the latter could potentially mislead customers.

**Review by zeeshan-devrev — CHANGES_REQUESTED** [review]
Small change for screen name tracking.

File: fern/docs/pages/sdks/mobile/react-native/features.mdx (hunk 5 above)
Comment (bojan) [reply fern/docs/pages/sdks/mobile/react-native/features.mdx:328]: I'd leave this for now as it was named like that before, we can try to think of a better name and change the APIs as well.

**Review by zeeshan-devrev — APPROVED** [review]

**Review by bc-devrev — APPROVED** [review]
